### PR TITLE
Update testing_counts_1.Rmd

### PR DESCRIPTION
**Pull request by pxr687**
new code conversion without "op_ed"

### CODE REVIEW

File: source/testing_counts_1.Rmd [hunk 1]
```diff
@@ -306,24 +306,38 @@ results from the histogram, we can calculate the result by tacking on
 the following commands to the above program:
 
 
-<!--- Resampling stats file; Matlab syntax closeish -->
+`{python}
```
Comment (matthew-brett) [inline source/testing_counts_1.Rmd:309]:
```suggestion
```{python}
```

File: source/testing_counts_1.Rmd [hunk 2]
```diff
@@ -306,24 +306,38 @@ results from the histogram, we can calculate the result by tacking on
...
-' Convert to a proportion.
-PRINT mm
-' Print the results.
+# Add the two results together.
+m = j + k
 
-' Note: The file "fruitfly" on the Resampling Stats software disk contains
-' this set of commands.
+# Convert to a proportion.
+mm = m/n_trials
```
Comment (matthew-brett) [inline source/testing_counts_1.Rmd:320]:
```suggestion
mm = m / n_trials
```

PEP 8 spacing.

File: source/testing_counts_1.Rmd [hunk 3]
```diff
@@ -464,22 +478,80 @@ discussed earlier for four children and for ten cows. Conventional
...
-    COUNT a BETWEEN 101 206 b
-    SCORE b z
-END
-HISTOGRAM z
-COUNT z >=9 k
-DIVIDE k 15000 kk
-PRINT kk
-' **10 Calves**
+# for 10000 repeats
+for i in np.arange(n_trials):
```
Comment (matthew-brett) [inline source/testing_counts_1.Rmd:492]:
```suggestion
for i in range(n_trials):
```

We defaulted to `range` in the earlier notebooks.

File: source/testing_counts_1.Rmd [hunk 4]
```diff
@@ -464,22 +478,80 @@ discussed earlier for four children and for ten cows. Conventional
...
+
+    # generate 10 numbers between 1 and 206
+    a = rnd.integers(low = 1, high = 206, size = sample_size)
+
+    # count how many numbers were between 101 and 206
+    b = np.sum((a >= 101) & ((a <= 206)))
```
Comment (matthew-brett) [inline source/testing_counts_1.Rmd:498]:
```suggestion
    a = rnd.choice(['female', 'male'], 
        p=[p_female, 1 - p_female], 
        size = sample_size)
    b = np.sum(a == 'female')
```

File: source/testing_counts_1.Rmd [hunk 5]
```diff
@@ -464,22 +478,80 @@ discussed earlier for four children and for ten cows. Conventional
 methods are discussed for comparison in @sec-testing-procedures. Here is the
 resampling solution in RESAMPLING STATS:
 
+```{python}
+# set the number of trials
+n_trials = 10000
 
-<!--- Resampling stats file; Matlab syntax closeish -->
+# set the size of each sample
+sample_size = 10
```
Comment (matthew-brett) [inline source/testing_counts_1.Rmd:486]:
```suggestion
sample_size = 10

# Probability of any one calf being female.
p_female = 100 / 206
```

See below for where this is used.

If you do go down this route, would need harmonizing wi the text above.

File: source/testing_counts_1.Rmd [hunk 6]
```diff
@@ -464,22 +478,80 @@ discussed earlier for four children and for ten cows. Conventional
...
+    a = rnd.integers(low = 1, high = 206, size = sample_size)
+
+    # count how many numbers were between 101 and 206
+    b = np.sum((a >= 101) & ((a <= 206)))
+
+    # store the result of the current trial
+    scores[i] = b
+
+# plot a histogram of the scores
+plt.title(f"Number of females in {n_trials} samples of \n{sample_size} simulated calves")
```
Comment (matthew-brett) [inline source/testing_counts_1.Rmd:504]: We should make a note in `notes/general.md` that we need to have covered f-strings by this point.

File: source/testing_counts_1.Rmd [hunk 7]
```diff
@@ -464,22 +478,80 @@ discussed earlier for four children and for ten cows. Conventional
...
+plt.hist(scores)
+plt.xlabel('Number of Females')
+plt.ylabel('Frequency')
+plt.show()
+
+# count the number of scores that were greater than or equal to 9
+k = np.sum(scores >= 9)
+
+# express as a proportion
+kk = k/n_trials
```
Comment (matthew-brett) [inline source/testing_counts_1.Rmd:514]:
```suggestion
kk = k / n_trials
```

File: source/testing_counts_1.Rmd [hunk 8]
```diff
@@ -464,22 +478,80 @@ discussed earlier for four children and for ten cows. Conventional
...
+scores <- numeric(n_trials)
+
+# for 10000 repeats
+for (i in 1:n_trials) {
+
+    # generate 10 numbers between 1 and 206
+    a <- sample(1:206, size = sample_size)
+
+    # count how many numbers were between 101 and 206
+    b <- sum((a >= 101) & ((a <= 206)))
```
Comment (matthew-brett) [inline source/testing_counts_1.Rmd:537]: Consider adapting as above for Python code.

File: source/testing_counts_1.Rmd [hunk 9]
```diff
@@ -464,22 +478,80 @@ discussed earlier for four children and for ten cows. Conventional
...
+
+    # store the result of the current trial
+    scores[i] <- b
+}
+
+# plot a histogram of the scores
+title_of_plot <- paste0("Number of females in", n_trials, " samples of \n", sample_size, " simulated calves")
```
Comment (matthew-brett) [inline source/testing_counts_1.Rmd:544]: Add note in `notes/general.md` to make sure we've covered `paste` (? and `paste0`) by this point.

File: source/testing_counts_1.Rmd [hunk 10]
```diff
@@ -464,22 +478,80 @@ discussed earlier for four children and for ten cows. Conventional
...
+hist(scores, xlab = 'Number of Females', main = title_of_plot)
+
+# count the number of scores that were greater than or equal to 9
+k <- sum(scores >= 9)
+
+# express as a proportion
+kk <- k/n_trials
```
Comment (matthew-brett) [inline source/testing_counts_1.Rmd:551]:
```suggestion
kk <- k / n_trials
```

**Review by matthew-brett — COMMENTED** [review]
For your consideration.

File: source/testing_counts_1.Rmd (hunk 6 above)
Comment (pxr687) [reply source/testing_counts_1.Rmd:503]: Thanks - I will do that, I was thinking mostly of making the graph as informative as possible, but didn't consider the f-strings might be confusing for a reader.

File: source/testing_counts_1.Rmd (hunk 9 above)
Comment (pxr687) [reply source/testing_counts_1.Rmd:543]: Thanks, same comment as with f-strings above (nice to have the informative title, but didn't consider paste might be a bit confusing...).

File: source/testing_counts_1.Rmd [hunk 11]
```diff
@@ -464,22 +478,79 @@ discussed earlier for four children and for ten cows. Conventional
...
+
+    # store the result of the current trial
+    scores[i] = b
+
+# plot a histogram of the scores
+plt.title(f"Number of females in {n_trials} samples of \n{sample_size} simulated calves")
+plt.hist(scores)
+plt.xlabel('Number of Females')
+plt.ylabel('Frequency')
+plt.show()
```
Comment (matthew-brett) [inline source/testing_counts_1.Rmd:507]:
```suggestion
```

Should not be necessary in notebook - we don't use it elsewhere.

File: source/testing_counts_1.Rmd (hunk 8 above)
Comment (matthew-brett) [reply source/testing_counts_1.Rmd:536]: Happy to do this if you like?

File: source/testing_counts_1.Rmd (hunk 5 above)
Comment (matthew-brett) [reply source/testing_counts_1.Rmd:486]: Happy to make the changes to the text as well - do you want me to do that?

**Review by matthew-brett — COMMENTED** [review]
More suggestions.

**Comment by matthew-brett** [comment]
Peter - where are we with this one?

**Comment by matthew-brett** [comment]
I merged these in with more edits - closing here.

**Comment by matthew-brett** [comment]
Main merge commit was 350112e, with a few subsequent edits for comments above.

**Pull request closed without merging**